Assets/Scripts/Entity/Player/Sanity/SanityController.cs

```diff
@@ -16,64 +14,270 @@ public enum StatType { Health, Saturation, Hydration };
+
+            if (applyCampfireBonus)
+                CheckCampfireRange();
+        }
 
         /// <summary>
-        /// Sums up a tick weighting the total change rate with <see cref="severity"/>. 
-        /// Only sums up a positive tick if player doesn't have full sanity. Ticks are either
-        /// positive or negative 1. If summed up tick, changes player sanity.
+        /// Depending on the player's needs (health, saturation, hydration), sum up a tick 
+        /// weighting the total change rate with <see cref="severity"/>. 
+        /// Only sum up a positive tick if player doesn't have full sanity. Ticks are either
+        /// positive or negative 1. If summed up tick, change player sanity.
         /// </summary>
         private void TickSanity()
         {
-            _nextTick += sanityMath.GetCurrentChange();
-            if (_playerState.Sanity == 100 && _nextTick > 0)
+            _needsTick += sanityMath.GetCurrentChange();
+            if (_playerState.Sanity == 100 && _needsTick > 0)
             {
                 // don't build up a positive tick value when sanity == 100
-                _nextTick = 0;
+                _needsTick = 0;
                 return;
             }
-            if (_nextTick >= 1)
+            if (_needsTick >= 1)
             {
                 _playerState.ChangePlayerSanity(1);
-                _nextTick = 0;
+                _needsTick = 0;
+            }
+            else if (_needsTick <= -1)
+            {
+                if (!_pauseTick)
+                    _playerState.ChangePlayerSanity(-1);
+                _needsTick = 0;
+            }
+        }
+
+        /// <summary>
+        /// Set player's status to fighting as soon as he performs an attack.
+        /// </summary>
+        private void OnAttackPerformed()
+        {
+            if (!_isFighting)
+                _isFighting = true;
+            _enemy = _attackLogic.lastAttacked;
+        }
+
+        /// <summary>
+        /// Count up the hits performed on the player. If <see cref="criticalHitCount"/> is reached,
+        /// apply a hit malus.
+        /// Also make the player stay in / enter combat if he is being hit.
+        /// </summary>
+        /// <param name="attacker">The EnemyController of the attacker</param>
+        private void OnPlayerHit(EnemyController attacker)
+        {
+            // enter combat
+            if (!_isFighting)
+                _isFighting = true;
+
+            // get the EnemyController passed here, so that e.g. if you attack a non-aggressive,
+            // run away and then get hit by an aggressive (which you don't fight back/runaway), you don't get the non-aggressive malus
+            // also necessary to prevent nullref if not having attacked at all, but get hit
+            _enemy = attacker;
+
+            _hitCounter += 1;
+            if (_hitCounter >= criticalHitCount)
+            {
+                AddUpEventTick(hitMalus);
+                _hitCounter = 0;
```

Comment:
I think it would be cooler when the hit counter doesnt reset here, but that's up to you :D

Reply:
But you just get the malus one time.. is that what you mean?

Reply:
aso wenn man den malus nur einmal kriegen soll dann kann es ja so bleiben.

Reply:
Nein, wenn du das nicht resettest, kriegst du den malus nur einmal :)